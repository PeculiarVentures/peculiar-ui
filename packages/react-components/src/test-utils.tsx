import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
