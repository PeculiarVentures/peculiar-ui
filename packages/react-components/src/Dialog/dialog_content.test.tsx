import { renderWithWrapper as render } from '../test-utils';
import { DialogContent } from './index';

describe('<DialogContent />', () => {
  it('should render with default styles', () => {
    const { asFragment } = render(
      <DialogContent>
        Content
      </DialogContent>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass className', () => {
    const { asFragment } = render(
      <DialogContent className="my-class-name">
        Content
      </DialogContent>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render border', () => {
    const { asFragment } = render(
      <DialogContent dividers>
        Content
      </DialogContent>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render error', () => {
    const { asFragment } = render(
      <DialogContent error="Some API error">
        Content
      </DialogContent>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
