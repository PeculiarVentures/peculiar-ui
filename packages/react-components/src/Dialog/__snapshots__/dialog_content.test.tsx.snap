// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogContent /> should pass className 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-color: var(--pv-color-gray-5);
  border-style: solid;
  border-width: 1px 0px 1px 0px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  padding: 0 var(--pv-size-base-4);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.emotion-2 {
  overflow: hidden;
  -webkit-transition: height 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: height 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  height: 0;
  visibility: hidden;
  margin: 0 calc(var(--pv-size-base-3) * -1);
}

.emotion-3 {
  background: var(--pv-color-wrong-tint-5);
  border-radius: 4px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: var(--pv-size-base-2) var(--pv-size-base-4);
  box-sizing: border-box;
  margin: var(--pv-size-base-2) 0;
  padding: var(--pv-size-base-2) var(--pv-size-base-3);
}

.emotion-4 {
  margin: 0;
  color: var(--pv-color-wrong);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  padding: var(--pv-size-base) 0px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-height: var(--pv-text-b3-height);
}

.emotion-5 {
  overflow-y: auto;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  padding: var(--pv-size-base-3) var(--pv-size-base-4) var(--pv-size-base-6) var(--pv-size-base-4);
}

<div
    class="my-class-name emotion-0"
    data-key="dialog.content"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div>
          <div
            class="emotion-3"
            role="alert"
          >
            <p
              class="emotion-4"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="emotion-5"
    >
      Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`<DialogContent /> should render border 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-color: var(--pv-color-gray-5);
  border-style: solid;
  border-width: 1px 0px 1px 0px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  padding: 0 var(--pv-size-base-4);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.emotion-2 {
  overflow: hidden;
  -webkit-transition: height 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: height 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  height: 0;
  visibility: hidden;
  margin: 0 calc(var(--pv-size-base-3) * -1);
}

.emotion-3 {
  background: var(--pv-color-wrong-tint-5);
  border-radius: 4px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: var(--pv-size-base-2) var(--pv-size-base-4);
  box-sizing: border-box;
  margin: var(--pv-size-base-2) 0;
  padding: var(--pv-size-base-2) var(--pv-size-base-3);
}

.emotion-4 {
  margin: 0;
  color: var(--pv-color-wrong);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  padding: var(--pv-size-base) 0px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-height: var(--pv-text-b3-height);
}

.emotion-5 {
  overflow-y: auto;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  padding: var(--pv-size-base-3) var(--pv-size-base-4) var(--pv-size-base-6) var(--pv-size-base-4);
}

<div
    class="emotion-0"
    data-key="dialog.content"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div>
          <div
            class="emotion-3"
            role="alert"
          >
            <p
              class="emotion-4"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="emotion-5"
    >
      Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`<DialogContent /> should render error 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-color: var(--pv-color-gray-5);
  border-style: solid;
  border-width: 1px 0px 1px 0px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  padding: 0 var(--pv-size-base-4);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.emotion-2 {
  overflow: visible;
  -webkit-transition: height 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: height 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  height: auto;
  margin: 0 calc(var(--pv-size-base-3) * -1);
}

.emotion-3 {
  background: var(--pv-color-wrong-tint-5);
  border-radius: 4px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: var(--pv-size-base-2) var(--pv-size-base-4);
  box-sizing: border-box;
  margin: var(--pv-size-base-2) 0;
  padding: var(--pv-size-base-2) var(--pv-size-base-3);
}

.emotion-4 {
  margin: 0;
  color: var(--pv-color-wrong);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  padding: var(--pv-size-base) 0px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-height: var(--pv-text-b3-height);
}

.emotion-5 {
  overflow-y: auto;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  padding: var(--pv-size-base-3) var(--pv-size-base-4) var(--pv-size-base-6) var(--pv-size-base-4);
}

<div
    class="emotion-0"
    data-key="dialog.content"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div>
          <div
            class="emotion-3"
            role="alert"
          >
            <p
              class="emotion-4"
            >
              Some API error
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="emotion-5"
    >
      Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`<DialogContent /> should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-color: var(--pv-color-gray-5);
  border-style: solid;
  border-width: 1px 0px 1px 0px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  padding: 0 var(--pv-size-base-4);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.emotion-2 {
  overflow: hidden;
  -webkit-transition: height 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: height 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  height: 0;
  visibility: hidden;
  margin: 0 calc(var(--pv-size-base-3) * -1);
}

.emotion-3 {
  background: var(--pv-color-wrong-tint-5);
  border-radius: 4px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: var(--pv-size-base-2) var(--pv-size-base-4);
  box-sizing: border-box;
  margin: var(--pv-size-base-2) 0;
  padding: var(--pv-size-base-2) var(--pv-size-base-3);
}

.emotion-4 {
  margin: 0;
  color: var(--pv-color-wrong);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  padding: var(--pv-size-base) 0px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-height: var(--pv-text-b3-height);
}

.emotion-5 {
  overflow-y: auto;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  padding: var(--pv-size-base-3) var(--pv-size-base-4) var(--pv-size-base-6) var(--pv-size-base-4);
}

<div
    class="emotion-0"
    data-key="dialog.content"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div>
          <div
            class="emotion-3"
            role="alert"
          >
            <p
              class="emotion-4"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="emotion-5"
    >
      Content
    </div>
  </div>
</DocumentFragment>
`;
